luigi/scheduler.py

```diff
@@ -402,7 +398,7 @@ def prune(self, config):
 
     def get_tasks(self, state, *statuses):
         num_self_tasks = len(self.tasks)
-        num_state_tasks = sum(len(state._status_tasks[status]) for status in statuses)
+        num_state_tasks = len(list(state.get_active_tasks_by_status(*statuses)))
```

Comment:
is this just an optimization?

Reply:
Kinda! Mostly needed to change it since `scheduler.py` should not be accessing `state._status_tasks`